src/ethereum/ethNetwork.js

```diff
@@ -477,89 +477,17 @@ export class EthereumNetwork {
     }
   }
 
-  async fetchGet(url: string, _options: Object = {}) {
-    const options = { ..._options }
-    options.method = 'GET'
-    const response = await this.ethEngine.io.fetch(url, options)
-    if (!response.ok) {
-      const {
-        blockchairApiKey,
-        blockcypherApiKey,
-        etherscanApiKey,
-        bscscanApiKey,
-        ftmscanApiKey,
-        infuraProjectId,
-        polygonscanApiKey
-      } = this.ethEngine.initOptions
-      // remove API keys from error messages
-      if (typeof etherscanApiKey === 'string')
-        url = url.replace(etherscanApiKey, 'private')
-      if (Array.isArray(etherscanApiKey)) {
-        for (const key of etherscanApiKey) {
-          url = url.replace(key, 'private')
-        }
-      }
-      if (typeof polygonscanApiKey === 'string')
-        url = url.replace(polygonscanApiKey, 'private')
-      if (Array.isArray(polygonscanApiKey)) {
-        for (const key of polygonscanApiKey) {
-          url = url.replace(key, 'private')
-        }
-      }
-      if (typeof bscscanApiKey === 'string')
-        url = url.replace(bscscanApiKey, 'private')
-      if (Array.isArray(bscscanApiKey)) {
-        for (const key of bscscanApiKey) {
-          url = url.replace(key, 'private')
-        }
-      }
-      if (blockchairApiKey) url = url.replace(blockchairApiKey, 'private')
-      if (blockcypherApiKey) url = url.replace(blockcypherApiKey, 'private')
-      if (ftmscanApiKey) url = url.replace(ftmscanApiKey, 'private')
-      if (infuraProjectId) url = url.replace(infuraProjectId, 'private')
-      this.throwError(response, 'fetchGet', url)
```

Comment:
Removing this ugly code means printing api keys to the console or are being shown to the user, depending on how the throw from throwError is handled.